tests/Rules.Framework.Tests/RulesEngineTests.cs

```diff
@@ -59,6 +61,92 @@ public async Task AddRuleAsync_GivenEmptyRuleDataSource_AddsRuleSuccesfully()
+            {
+                ContentContainer = new ContentContainer<ContentType>(contentType, (t) => new object()),
+                DateBegin = dateBegin,
+                DateEnd = dateEnd,
+                Name = "Rule 1",
+                Priority = 3,
+                RootCondition = new StringConditionNode<ConditionType>(ConditionType.IsoCountryCode, Operators.Equal, "USA")
+            };
+
+            Rule<ContentType, ConditionType> rule2 = new Rule<ContentType, ConditionType>
+            {
+                ContentContainer = new ContentContainer<ContentType>(contentType, (t) => new object()),
+                DateBegin = new DateTime(2020, 01, 01),
+                DateEnd = new DateTime(2021, 01, 01),
+                Name = "Rule 2",
+                Priority = 200,
+                RootCondition = new StringConditionNode<ConditionType>(ConditionType.IsoCountryCode, Operators.Equal, "USA")
+            };
+
+            Rule<ContentType, ConditionType> rule3 = new Rule<ContentType, ConditionType>
+            {
+                ContentContainer = new ContentContainer<ContentType>(contentType, (t) => new object()),
+                DateBegin = dateBegin,
+                DateEnd = dateEnd,
+                Name = "Rule 3",
+                Priority = 1,
+                RootCondition = new StringConditionNode<ConditionType>(ConditionType.IsoCountryCode, Operators.Equal, "CHE")
+            };
+
+            IEnumerable<Rule<ContentType, ConditionType>> rules = new[]
+            {
+                rule1,
+                rule3
+            };
+
+            EvaluationOptions evaluationOptions = new EvaluationOptions
+            {
+                MatchMode = MatchModes.Exact
+            };
+
+            var expectedCondtionTypes = new List<ConditionType>() { ConditionType.IsoCountryCode };
+
+            Mock<IConditionTypeExtractor<ContentType, ConditionType>> mockCondtionTypeExtractor = new Mock<IConditionTypeExtractor<ContentType, ConditionType>>();
+            mockCondtionTypeExtractor.Setup(x => x.GetConditionTypes(It.IsAny<IEnumerable<Rule<ContentType, ConditionType>>>()))
+                .Returns(expectedCondtionTypes);
+
+            Mock<IRulesDataSource<ContentType, ConditionType>> mockRulesDataSource = SetupMockForRulesDataSource(rules);
+
+            Mock<IConditionsEvalEngine<ConditionType>> mockConditionsEvalEngine = SetupMockForConditionsEvalEngine((rootConditionNode, inputConditions, evalOptions) =>
+            {
+                switch (rootConditionNode)
+                {
+                    case StringConditionNode<ConditionType> stringConditionNode:
+                        return stringConditionNode.Operand == "USA";
+
+                    default:
+                        return false;
+                }
+            }, evaluationOptions);
+
+            IValidatorProvider validatorProvider = Mock.Of<IValidatorProvider>();
```

Comment:
It would be better to initialize all the mocks on the test constructor and then make the adjustments/setup on each test? Also the ` RulesEngine<ContentType, ConditionType> sut` can be initialized on the constructor and then we only need to call the ACT method. This is a suggestion to shorten the test line size and improve readability. If you agree, I can help do it otherwise, we can make this improvement in another feature

Reply:
Ok, I see no problem. I've made the changes in the class